scipy-1.0/minimize.tex

```diff
@@ -0,0 +1,118 @@
+    \hline
+    It is a wrapper? &  &  & & &  &  & & & \cmark \\
+    \hline
+    \nth{1} derivatives &  & & \cmark & \cmark & \cmark & \cmark & \cmark & \cmark & \cmark \\
+    \hline
+    \nth{2} derivatives &  &  &  & Approx. & \cmark & \cmark & \cmark & \cmark & \cmark \\
+    \hline
+    \makecell{Iterative Hessian \\
+    factorization} & & & & & \cmark &  & \cmark &  & \cmark \\
+    \hline
+    Local convergence& & & L & S & S & Q & S & Q & ?  \\
+    \hline
+    Global convergence & & &  & \cmark & \cmark & \cmark & \cmark & \cmark & ?  \\
+    \hline
+    \makecell{Line-search (LS) or\\ trust-region (TR)} & Neither  & LS & LS & LS & LS & TR & TR & TR & TR \\
+    \hline
+    References & \cite{nelder_simplex_1965, wright_direct_1996} & \cite{powell_efficient_1964} &
+      \cite{polak_note_1969, nocedal_numerical_2006} & \cite{nocedal_numerical_2006} & \cite{nocedal_numerical_2006}  &
+      \cite{powell_new_1970, nocedal_numerical_2006} &  \cite{steihaug_conjugate_1983, nocedal_numerical_2006} &
+      \cite{conn_trust_2000, more_computing_1983} & \cite{gould_solving_1999, lenders_trlib:_2016} \\
+    \hline
+  \end{tabular}
+  \caption{Optimization methods from \texttt{minimize} for unconstrained minimization.  The field \textit{It is a wrapper?}
+    informs if the implementation available in SciPy is a wrapper to a function written in other programming language
+    (e.g. C or FORTRAN). The fields \textit{\nth{1}} and \textit{\nth{2} derivatives}
+    informs if the first or second order derivatives are required. When \textit{\nth{2} derivatives} is flagged
+    with the word \textit{Approx.} it means that, while the algorithm does not requires the second-order derivatives from
+    the user, it computes an approximation internally and uses it to accelerate the method convergence.
+    \textit{Local convergence} gives
+    a lower bound to the rate of convergence of the iterations sequence once the iterate is close enough to the
+    objective function solution. The convergence rates mentioned at the table
+    are, from the slowest to the fastest:  linear (L), superlinear (S) and quadratic (Q). The \textit{global convergence}
+    is marked for the algorithms for which there are some proof it always converge to some stationary
+    point (i.e. a point $x^*$ for which $\nabla f(x^*) = 0$).}
+  \label{tab:minimize-unconstrained}
+\end{table}
+
+
+The \texttt{minimize} function provides methods for solving general non-linear optimization
+problems. It unifies several methods under the same general interface in order to facilitate
+interchanging between solvers with different characteristics.  Table~\ref{tab:minimize-unconstrained} summarizes
+all the methods available for solving unconstrained minimization problems of the type:
+\begin{equation}
+  \label{eq:unconstrained-minimization-prob}
+  \min_x f(x)
+\end{equation}
+where $f$ is a multivariate function $f: \mathbb{R}^n \rightarrow \mathbb{R}$ and $x \in \mathbb{R}^n$.
```

Comment:
I suggest we add the command \RR for \mathbb{R} as this will be used all over.
Also, to indicate multivariate/vector valued entities, do we want to use boldface? I did, but I can take it out. In any case, we should decide now.

Reply:
I borrow the notation from Nocedal book, which is the reference used for most of the methods.
He opts to not use boldface for vectors. I think 

I think I prefer this notation as it is, it helps keeping things simple. But I think we should wait and see how the remaining sections will choose to deal with vectors.